src/common/arrow/arrow_row_batch.cpp

```diff
@@ -106,82 +164,179 @@ void ArrowRowBatch::templateCopyNonNullValue<bool>(
+    std::int64_t propertyId = 2;
+    for (auto& [name, val] : value->nodeVal->getProperties()) {
+        appendValue(vector->childData[propertyId].get(), *typeInfo.childrenTypesInfo[propertyId],
+            val.get());
+        propertyId++;
+    }
+}
+
+template<>
+void ArrowRowBatch::templateCopyNonNullValue<REL>(
+    ArrowVector* vector, const main::DataTypeInfo& typeInfo, Value* value, std::int64_t pos) {
+    appendValue(vector->childData[0].get(), *typeInfo.childrenTypesInfo[0],
+        value->relVal->getSrcNodeIDVal());
+    appendValue(vector->childData[1].get(), *typeInfo.childrenTypesInfo[1],
+        value->relVal->getDstNodeIDVal());
+    std::int64_t propertyId = 2;
+    for (auto& [name, val] : value->relVal->getProperties()) {
+        appendValue(vector->childData[propertyId].get(), *typeInfo.childrenTypesInfo[propertyId],
+            val.get());
+        propertyId++;
+    }
+}
+
+void ArrowRowBatch::copyNonNullValue(
+    ArrowVector* vector, const main::DataTypeInfo& typeInfo, Value* value, std::int64_t pos) {
+    switch (typeInfo.typeID) {
     case BOOL: {
-        templateCopyNonNullValue<bool>(vector, value, pos);
+        templateCopyNonNullValue<BOOL>(vector, typeInfo, value, pos);
     } break;
     case INT64: {
-        templateCopyNonNullValue<std::int64_t>(vector, value, pos);
+        templateCopyNonNullValue<INT64>(vector, typeInfo, value, pos);
     } break;
     case DOUBLE: {
-        templateCopyNonNullValue<std::double_t>(vector, value, pos);
+        templateCopyNonNullValue<DOUBLE>(vector, typeInfo, value, pos);
     } break;
     case DATE: {
-        templateCopyNonNullValue<date_t>(vector, value, pos);
+        templateCopyNonNullValue<DATE>(vector, typeInfo, value, pos);
     } break;
     case TIMESTAMP: {
-        templateCopyNonNullValue<timestamp_t>(vector, value, pos);
+        templateCopyNonNullValue<TIMESTAMP>(vector, typeInfo, value, pos);
     } break;
     case INTERVAL: {
-        templateCopyNonNullValue<interval_t>(vector, value, pos);
+        templateCopyNonNullValue<INTERVAL>(vector, typeInfo, value, pos);
     } break;
     case STRING: {
-        templateCopyNonNullValue<std::string>(vector, value, pos);
+        templateCopyNonNullValue<STRING>(vector, typeInfo, value, pos);
+    } break;
+    case LIST: {
+        templateCopyNonNullValue<LIST>(vector, typeInfo, value, pos);
+    } break;
+    case NODE_ID: {
+        templateCopyNonNullValue<NODE_ID>(vector, typeInfo, value, pos);
+    } break;
+    case NODE: {
+        templateCopyNonNullValue<NODE>(vector, typeInfo, value, pos);
+    } break;
+    case REL: {
+        templateCopyNonNullValue<REL>(vector, typeInfo, value, pos);
     } break;
     default: {
         throw RuntimeException("Data type " + Types::dataTypeToString(value->dataType) +
-                               " is not supported for arrow exportation.");
+                               " is not supported for arrow exportation. in copyNonNullValue");
```

Comment:
`is not supported for arrow exportation in copyNonNullValue.`

Reply:
Reworked this exception message.